### PR TITLE
Adds more plan functionality

### PR DESCRIPTION
**Pull request by ktayah**
Added the find, update, and create functions under `Braintree.Plan` to access more of the Plan API.

Edit: Also added delete function

### CODE REVIEW

File: test/integration/plan_test.exs [hunk 1]
```diff
@@ -13,4 +13,39 @@ defmodule Braintree.Integration.PlanTest do
...
+    {:ok, [%Plan{id: existing_plan_id} | _]} = Plan.all()
+    {:ok, %Plan{} = plan} = Plan.find(existing_plan_id)
+
+    assert plan
+    assert plan.id == existing_plan_id
+  end
+
+  # This relies on there being at least one plan in the sandbox account.
+  test "update/2 updates a plan" do
+    {:ok, [%Plan{id: existing_plan_id} | _]} = Plan.all()
```
Comment (sorentwo) [inline test/integration/plan_test.exs:44]: Maybe switch to creating a plan as part of the setup and then reference it?

File: test/integration/plan_test.exs [hunk 2]
```diff
@@ -13,4 +13,39 @@ defmodule Braintree.Integration.PlanTest do
     assert plan.id
     assert plan.name
   end
+
+  # This will create a new plan every time it's ran, make sure to cleanup if ran excessively
```
Comment (sorentwo) [inline test/integration/plan_test.exs:17]: Ideally we'd delete them afterwards to clean up. I'm not sure what the plan limit is, and I'm sure we don't want to clean them out manually.

Comment (ktayah) [reply test/integration/plan_test.exs:17]: I agree, ideally I can delete and clear them out manually as well. (Would also be helpful for preloading data for testing `Plan.all()`) 

But it seems that a DELETE request on "plans/#{plan_id}" works properly but returns html due to it probably being something expected to be used from within the web platform, which breaks the XML decoder. I included some code below to show what I mean:
```
iex(2)> Braintree.HTTP.delete("plans/p8bg")
"<html><body>You are being <a href=\"https://api.sandbox.braintreegateway.com/merchants/2ww49grjxnbgqwb6/plans\">redirected</a>.</body></html>"
{:ok, :ok}
[error] unprocessable response
```

Maybe we just add a private function in `plan_test.exs` to handle deleting post and ignore the failing decoding?

Comment (sorentwo) [reply test/integration/plan_test.exs:17]: Ah, they redirect after a delete request. The response should have a 30X response code. Can you modify the HTTP client to handle redirects differently? Don't bother parsing the response and return an `:ok` tuple or empty binary instead?

File: test/integration/plan_test.exs [hunk 3]
```diff
@@ -5,12 +5,78 @@ defmodule Braintree.Integration.PlanTest do
...
+        currency_iso_code: "USD",
+        price: "10.00"
+      })
+
+    assert plan
+    assert plan.id
+    assert plan.name == "testing_create_plan"
+    assert plan.price == "10.00"
+
+    :ok = Plan.delete(plan.id)
```
Comment (sorentwo) [inline test/integration/plan_test.exs:46]:
```suggestion
    assert :ok = Plan.delete(plan.id)
```

**Review by sorentwo — APPROVED** [review]

**Pull request merged**